etcdctl/command/cluster_health.go

```diff
@@ -51,27 +50,43 @@ func handleClusterHealth(c *cli.Context) {
 	time.Sleep(time.Second)
 
 	// are all the members makeing progress?
-	_, ls1, err := getLeaderStats(tr, []string{ep})
+	_, rs1, err := getLeaderStatus(tr, []string{ep})
 	if err != nil {
 		fmt.Println("cluster is unhealthy")
 		os.Exit(1)
 	}
 
-	fmt.Println("cluster is healthy")
-	// self is healthy
+	if rs1.Commit > rs0.Commit {
+		fmt.Printf("cluster is healthy: raft is making progress [commit index: %v->%v]\n", rs0.Commit, rs1.Commit)
+	} else {
+		fmt.Printf("cluster is unhealthy: raft is not making progress [commit index: %v]\n", rs0.Commit)
+	}
+	fmt.Printf("leader is %v\n", rs0.Lead)
+
 	var prints []string
 
-	prints = append(prints, fmt.Sprintf("member %s is healthy\n", ls1.Leader))
-	for name, fs0 := range ls0.Followers {
-		fs1, ok := ls1.Followers[name]
+	for id, pr0 := range rs0.Progress {
+		pr1, ok := rs1.Progress[id]
 		if !ok {
 			fmt.Println("Cluster configuration changed during health checking. Please retry.")
 			os.Exit(1)
 		}
-		if fs1.Counts.Success <= fs0.Counts.Success {
-			prints = append(prints, fmt.Sprintf("member %s is unhealthy\n", name))
+		if pr1.Match <= pr0.Match {
+			var msg string
+			if pr0.State == pr1.State {
+				msg = fmt.Sprintf("member %s is unhealthy because its match index is not catching up with commit index [match: %v->%v, next: %v->%v, state: %v]\n", id, pr0.Match, pr1.Match, pr0.Next, pr1.Next, pr1.State)
+			} else {
+				msg = fmt.Sprintf("member %s is unhealthy because its match index is not catching up with commit index [match: %v->%v, next: %v->%v, state: %v->%v]\n", id, pr0.Match, pr1.Match, pr0.Next, pr1.Next, pr0.State, pr1.State)
+			}
+			prints = append(prints, msg)
 		} else {
-			prints = append(prints, fmt.Sprintf("member %s is healthy\n", name))
+			var msg string
+			if pr0.State == pr1.State {
+				msg = fmt.Sprintf("member %s is healthy [match: %v->%v, next: %v->%v, state: %v]\n", id, pr0.Match, pr1.Match, pr1.Next, pr1.Next, pr1.State)
```

Comment:
can we make this consistent with the previous one? member %s is healthy: raft is making progress [match ...] (we do not need next and state)